### PR TITLE
Reduce the cleanup time before/after each test

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Use for namespace cleanup a rest client with a higher rate limit and
operator on both namespaces which need to be cleaned in parallel.

Reduce to cleanup time from 5-6 seconds to 1-2 seconds.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Still wip to see if unexpected consequences arise form the increased limit.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dhiller** [comment]
@rmohr looks fishy, sig-compute lanes are failing on the same test?

**Comment by rmohr** [comment]
seems like the faster cleanup  reveals a bug in this test:

```
Tests Suite: [sig-compute]HookSidecars [rfe_id:2667][crit:medium][vendor:cnv-qe@redhat.com][level:component] VMI definition with SM BIOS hook sidecar should not start with hook sidecar annotation when the version is not provided
```

**Comment by rmohr** [comment]
> @rmohr looks fishy, sig-compute lanes are failing on the same test?

Added a fix. There was an issue in the tests which got revealed.

**Comment by dhiller** [comment]
@rmohr strange, here the same failure in two compute test lanes

```
 panic: migrationpolicies.migrations.kubevirt.io "testpolicy" not found
```

Example:

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7119/pull-kubevirt-e2e-k8s-1.22-sig-compute/1484236546046431232#1:build-log.txt%3A3426

**Comment by rmohr** [comment]
> @rmohr strange, here the same failure in two compute test lanes

Yes, the cleanup was wrong. Fixed.

**Comment by rmohr** [comment]
Interesting: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7119/pull-kubevirt-e2e-k8s-1.21-sig-compute/1484443290790006784.

 Seems like something is happeninge before the failing test which deletes the operator pod. The pod is in all failing lanes only 4 or 5 seconds old and not yet ready.

**Comment by rmohr** [comment]
/cc @dhiller 

can you have another look? Tests look good now. But retriggering all to get more data points.

**Comment by rmohr** [comment]
/test pull-kubevirt-e2e-kind-1.19-vgpu


Failed after the test run on the cluster destroy, so unrelated: 
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7119/pull-kubevirt-e2e-kind-1.19-vgpu/1488062138487410688

File: tests/utils.go [hunk 1]
```diff
@@ -1805,135 +1806,140 @@ func GetContainerOfPod(pod *k8sv1.Pod, containerName string) *k8sv1.Container {
...
-		_, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
-		if err != nil {
-			continue
-		}
-
-		// Clean namespace labels
-		err = libnet.RemoveAllLabelsFromNamespace(virtCli, namespace)
-		util2.PanicOnError(err)
+			// Clean namespace labels
+			err = libnet.RemoveAllLabelsFromNamespace(virtCli, namespace)
```
Comment (dhiller) [inline tests/utils.go:1823]: nit: can we stick to the below style, i.e. just directly consuming the result of the action, or is there a reason for the declaration?

File: tests/utils.go [hunk 2]
```diff
@@ -1805,135 +1806,140 @@ func GetContainerOfPod(pod *k8sv1.Pod, containerName string) *k8sv1.Container {
...
 
-		_, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
-		if err != nil {
-			continue
-		}
-
-		// Clean namespace labels
-		err = libnet.RemoveAllLabelsFromNamespace(virtCli, namespace)
-		util2.PanicOnError(err)
+			// Clean namespace labels
```
Comment (dhiller) [inline tests/utils.go:1822]: nit: IMHO this and all the following comments are redundant, maybe we can just remove them?

**Review by dhiller — COMMENTED** [review]
Hey, sorry for the delay. Great to have the cleanup speed up! One question inside, besides that nits only.

File: tests/utils.go [hunk 3]
```diff
@@ -1805,135 +1806,140 @@ func GetContainerOfPod(pod *k8sv1.Pod, containerName string) *k8sv1.Container {
...
+	virtCli, err := client.TestClientWithHighRateLimits()
 	util2.PanicOnError(err)
 
+	wg := sync.WaitGroup{}
+	wg.Add(len(TestNamespaces))
 	for _, namespace := range TestNamespaces {
+		go func(namespace string) {
+			defer wg.Done()
+			_, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/utils.go:1818]: I agree. Shouldn't we check for not fount?

**Review by xpivarc — COMMENTED** [review]
/approve

@rmohr Do you want to continue with this work?

File: tests/utils.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/utils.go:1818]: Yes makes sense. 👍

**Comment by rmohr** [comment]
> @rmohr Do you want to continue with this work?

In general yes. But I don't expect to find the time soon to work on it again. Happy if anyone wants to take it over in the meantime. I think I wanted to wait for some more migration test fixes to be in to be sure that the result would really be stable enough to get merged.

**Comment by rmohr** [comment]
The fixes mentioned in the PR description were extracted and merged with https://github.com/kubevirt/kubevirt/pull/7169.

**Pull request closed without merging**